src/core/textures/Texture.js

```diff
@@ -216,15 +224,17 @@ export default class Texture extends EventEmitter
     {
         if (this.baseTexture)
         {
-            if (destroyBase)
+            // remove BaseTexture stored under the texture name if destroyBase is true or
+            // is our base texture (in which case this is the event callback for 'destroy')
+            if ((destroyBase === true || destroyBase === this.baseTexture)
```

Comment:
Do you often reuse a Texture that was generated from `Texture.fromImage()` after destroying its BaseTexture? This is primarily to take care of the textures generated automatically from Spritesheets and `Texture.fromImage()`, `Texture.fromLoader()`, etc. I feel that reusing the Texture instances in those cases is not a common (or even uncommon) use case.